aws_lambda_powertools/logging/types.py

```diff
@@ -0,0 +1,10 @@
+from typing import Any, Dict, Iterable, Union
+
+from typing_extensions import TypedDict
+
+class TypedLog(TypedDict, total=False):
+    level: Union[str, int]
+    location: str
+    timestamp: Union[str, int]
+    service: str
+    message: Union[Dict[str, Any], str, bool, Iterable]
```

Comment:
```suggestion
    message: Dict[str, Any] | str | bool | Iterable
```

see how much nicer it becomes after you bring in `from __future__ import annotations` I suggested earlier?

I've replaced `Iterable` for `List` as Iterable encompass a wider type of data structures that JSON doesn't, for example a `Set` is Iterable but would be serialized as a string.


```python
from aws_lambda_powertools import Logger

logger = Logger(service="sample")

item_set = {"banana", "apple"}  # this is set()
logger.info(item_set)
```

The output would be more or less like this

```json
{"level":"INFO","location":"<module>:19","message":"{'banana', 'apple'}","timestamp":"2023-06-08 10:50:34,451+0200","service":"sample"}
```